src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1199,6 +1199,12 @@ ImmutableArray<AttributeData> IMethodSymbol.GetReturnTypeAttributes()
         /// </summary>
         internal virtual void AddSynthesizedReturnTypeAttributes(ref ArrayBuilder<SynthesizedAttributeData> attributes)
         {
+            if (this.ReturnsByRefReadonly)
+            {
+                // PROTOTYPE(readonlyRefs) it is optional now as it will be generated in the next PR
+                var constructor = WellKnownMember.System_Runtime_CompilerServices_ReadOnlyAttribute__ctor;
+                AddSynthesizedAttribute(ref attributes, this.DeclaringCompilation.TrySynthesizeAttribute(constructor, isOptionalUse: true));
```

Comment:
From what I saw, The `CSharpCompilation.SynthesizeXXX()` methods are there to handle cases where some computation or checking is needed before the attribute is synthesized, otherwise, it returns null.
In this case, we don't do anything, but generating the attribute. The most obvious counter-example I found was `SynthesizeDebuggableAttribute()`.
Do we want to create a helper method for each well known attribute, even if it is a one-liner?

---
In reply to: [110279349](https://github.com/dotnet/roslyn/pull/18358#discussion_r110279349) [](ancestors = 110279349)